### DISCUSSION TITLE
0063 XLS-63d: SignIn Transaction

### DISCUSSION THREAD

**Discussion by dangell7** [XRPLF/XRPL-Standards#187] [XLS Proposals]
## The earlier version can be found by looking at the edit history.

**Comment by desimmons** [comment]
Is it possible that the arbitrary piece of data the user is being asked to sign is itself a transaction object?

**Reply by dangell7** [reply]
Good point, I could block anything starting with `1200` however submitting the tx_blob with this signature would result in a failure because the signature would not match the payload.

**Reply by desimmons** [reply]
I think a key feature of this transaction type should be that when creating it, the signer is guaranteed that no ledger modifications can be made. Not sure what the best way to achieve that is though

**Reply by ximinez** [reply]
A signature for a wrapper even around a fully valid transaction can't be a valid signature for that transaction. As long as there is any modification to the transaction, you'll get a completely different signature.

**Comment by shortthefomo** [comment]
This seems to be a common theme some of the more advanced wallets love to use, limiting what can be signed. eg Xaman if not mistaken also won't sign arbitrary random data as well.

Allowing this does then ask the question as to what's next?

But see the utility.

**Comment by brettmollin** [comment]
I see apps using an empty account_set transaction for this purpose.
